guides/micronaut-data-r2dbc-repository/micronaut-data-r2dbc-repository.adoc

```diff
@@ -0,0 +1,151 @@
+
+|===
+
+=== Controller
+
+common:validation.adoc[]
+
+Create a class to encapsulate the Update operations:
+
+source:GenreUpdateCommand[]
+
+callout:introspected[1]
+
+Create `GenreController`, a controller that exposes a resource with the common CRUD operations:
+
+source:GenreController[]
+
+callout:controller[number=1,arg0=/genres]
+callout:constructor-di[number=2,arg0=GenreRepository]
+<3> Maps a `GET` request to `/genres/{id}`, which attempts to show a genre. This illustrates the use of a URL path variable.
+<4> Returning an empty Reactor `Mono` when the genre doesn't exist makes the Micronaut framework respond with 404 (not found).
+<5> Maps a `PUT` request to `/genres`, which attempts to update a genre.
+<6> Adds `@Valid` to any method parameter that requires validation. Use a POJO supplied as a JSON payload in the request to populate command.
+<7> The `Mono.thenReturn(..)` method is used to return a `NO_CONTENT` response only if the update was successful. Headers are easily added by the `HttpResponse` API.
+<8> Maps a `GET` request to `/genres/list`, which returns a list of genres. 
+<9> You can bind `Pageable` as a controller method argument. Check the examples in the following test section and read the https://micronaut-projects.github.io/micronaut-data/latest/guide/configurationreference.html#io.micronaut.data.runtime.config.DataConfiguration.PageableConfiguration[Pageable configuration] options. For example, you can configure the default page size with the configuration property `micronaut.data.pageable.default-page-size`.
+<10> Maps a `POST` request to `/genres`, which attempts to save a genre. This example uses the `Mono.map(..)` method to obtain the result of the call to `save(String name)` and map it to a HTTP `CREATED` response if the save operation was successful. 
+<11> Maps a `POST` request to `/ex`, which generates an exception. This example demonstrates how to handle exceptions with `Mono.onErrorReturn(..)` which simply returns a `NO_CONTENT` response if an error occurs. More complex use cases (like logging the exception) can be handled with `Mono.onErrorResume(..)`.
+<12> Maps a `DELETE` request to `/genres/{id}`, which attempts to remove a genre. This illustrates the use of a URL path variable. The `Mono.then()` method is called to discard the result and return an empty `Mono` only if the call to `deleteById` is successful.
+
+=== Writing Tests
+
+Create a test to verify the CRUD operations:
+
+test:GenreControllerTest[]
+
+callout:micronaut-test[1]
+callout:http-client[2]
+callout:http-request[3]
+<4> If you care just about the object in the response, use `retrieve`.
+<5> Sometimes, receiving just the object is not enough, and you need information about the response. In this case, instead of `retrieve`, use the `exchange` method.
+
+common:testApp.adoc[]
+
+== Running the Application
+
+Execute the following command to run a MySQL container:
```

Comment:
this isn't needed when using test resources, right?

Reply:
resolved via https://github.com/micronaut-projects/micronaut-guides/pull/965/commits/700018c3687300494cb8b351be6f7b7401130268